src/Bootstrappers/MailTenancyBootstrapper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Illuminate\Config\Repository;
+use Illuminate\Support\Arr;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class MailTenancyBootstrapper implements TenancyBootstrapper
+{
+    protected array $originalConfig = [];
+
+    /**
+     * Tenant properties to be mapped to config (similarly to the TenantConfig feature).
+     *
+     * For example:
+     * [
+     *     'property' => 'config.key.name',
+     * ]
+     */
+    public static array $credentialsMap = [
+        'smtp_transport' => 'mail.mailers.smtp.transport',
+        'smtp_host' => 'mail.mailers.smtp.host',
+        'smtp_port' => 'mail.mailers.smtp.port',
+        'smtp_encryption' => 'mail.mailers.smtp.encryption',
+        'smtp_username' => 'mail.mailers.smtp.username',
+        'smtp_password' => 'mail.mailers.smtp.password',
+        'smtp_timeout' => 'mail.mailers.smtp.timeout',
+        'smtp_local_domain' => 'mail.mailers.smtp.local_domain',
+    ];
+
+    public function __construct(protected Repository $config)
+    {
+    }
+
+    public function bootstrap(Tenant $tenant): void
+    {
+        foreach (static::$credentialsMap as $storageKey => $configKey) {
+            /** @var Tenant&Model $tenant */
+            $override = Arr::get($tenant, $storageKey);
+
+            if (! is_null($override)) {
+                if (is_array($configKey)) {
```

Comment:
Do we have a use case for this?

Reply:
Good point, we don't. I think we can probably omit the whole `if (is_array($configKey))`.

Reply:
https://github.com/archtechx/tenancy/pull/989/commits/d895a30c30a65c7d875ed0f5a7ac5b14c8c3c038

Reply:
I think simplifying the copied code is fine here – we need to 1) set the config values in `bootstrap`, and 2) revert the config values to the original ones in `revert`. I changed a small thing in the bootstrap method (https://github.com/archtechx/tenancy/pull/989/commits/04ada13584a4bbf1efea1dbe049bb4c8bd8d40a8), and now, I don't think I'd change anything else here